packages/@ember/application/lib/application.js

```diff
@@ -483,10 +492,15 @@ const Application = Engine.extend({
     @method waitForDOMReady
   */
   waitForDOMReady() {
-    if (!this.$ || this.$.isReady) {
+    if (!hasDOM || this.rootDocument.readyState !== 'loading') {
```

Comment:
If the defaulting above already checked `hasDOM`, we can probably just check for `this._document === null` here